includes/classes/SiteAutomation/Request.php

```diff
@@ -81,7 +81,7 @@ public function get( $page, $widget ) {
 		$this->status = $this->get_status();
 		$site_automation_data = get_option( "sophi_site_automation_data_{$page}_{$widget}" );
 
-		if ( ! empty( $this->status['success'] ) && $site_automation_data ) {
+		if ( ! empty( $this->status['success'] ) && $site_automation_data && ( defined( 'SOPHI_BYPASS_CACHE' ) && ! SOPHI_BYPASS_CACHE ) ) {
```

Comment:
SOPHI_BYPASS_CACHE doesn't bypass data completely: if `is_wp_error( $response )` it will still return $site_automation_data [on line 101](https://github.com/globeandmail/sophi-for-wordpress/blob/873af133e12afb02b0d749c0ed4e66080a57f424/includes/classes/SiteAutomation/Request.php#L101) if that is available. To prevent that from happening, perhaps first define `$site_automation_data = false` and then fetch using get_option(...) only if `! SOPHI_BYPASS_CACHE`?

```
$site_automation_data = false;

if ( ! defined( 'SOPHI_BYPASS_CACHE' ) || ! SOPHI_BYPASS_CACHE ) {
	$site_automation_data = get_option( "sophi_site_automation_data_{$page}_{$widget}" );	
}

if ( $site_automation_data && ! empty( $this->status['success'] ) ) {
	return $site_automation_data;
}
```

Reply:
Good call @barryceelen I've adjusted here: https://github.com/globeandmail/sophi-for-wordpress/pull/183/commits/749241eefde03d05305be3a03ad51fe687de6ef5

Reply:
Yay!

Perhaps nitpicking: While the approach here works for the immediate need of bypassing cache in this function, from an architectural point of view I would expect setting a global variable to disable cache would also prevent the plugin from creating those caches and as well as retrieving them.

One fairly straightforward approach could be to introduce something along the lines of `sophi_cache_set( .... )` and `sophi_cache_get( ... )` functions, in stead of directly using the get_option() and set_option() (or ..transient() ) functions like the plugin currently does, and check the bypass cache global in those functions in stead. (That would also provide a nice point to introduce a filter that could decide to disable cache for a specific page/widget combination, or all of them, in stead of using a global that always disables all of them) @jeffpaul for thoughts.

Reply:
I do agree that would be nice to have, though wondering how would it fit with the schedule and if leaving this for now would have a huge impact in the future for people using that constant if we go the filter approach as well @felipeelia